datalab/datalab_session/tests/test_operations.py

```diff
@@ -0,0 +1,196 @@
+            ],
+            'input_number': 1,
+            'input_string': 'test'
+        })
+
+        # in order files
+        input_data = {
+            'input_files': [
+                {'basename': 'file1'},
+                {'basename': 'file2'}
+            ],
+            'input_number': 1,
+            'input_string': 'test'
+        }
+        normalized_input_data = self.data_operation._normalize_input_data(input_data)
+        self.assertEqual(normalized_input_data, {
+            'input_files': [
+                {'basename': 'file1'},
+                {'basename': 'file2'}
+            ],
+            'input_number': 1,
+            'input_string': 'test'
+        })
+
+        # missing files
+        input_data = {
+            'input_number': 1,
+            'input_string': 'test'
+        }
+        normalized_input_data = self.data_operation._normalize_input_data(input_data)
+        self.assertEqual(normalized_input_data, {
+            'input_number': 1,
+            'input_string': 'test'
+        })
+      
+    def test_name(self):
+        self.assertEqual(self.data_operation.name(), 'SampleDataOperation')
+    
+    def test_description(self):
+        self.assertEqual(self.data_operation.description(), 'Testing class for DataOperation')
+    
+    def test_wizard_description(self):
+        wizard_description = {
+            'name': 'SampleDataOperation',
+            'description': 'Testing class for DataOperation',
+            'category': 'test',
+            'inputs': {
+                'input_files': {
+                    'name': 'Input Files',
+                    'description': 'The input files to operate on',
+                    'type': 'file',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_number': {
+                    'name': 'Input Number',
+                    'description': 'The input number to operate on',
+                    'type': 'number',
+                    'minimum': 1,
+                    'maximum': 999
+                },
+                'input_string': {
+                    'name': 'Input String',
+                    'description': 'The input string to operate on',
+                    'type': 'string',
+                    'minimum': 1,
+                    'maximum': 999
+                }
+            }
+        }
+        self.assertEqual(self.data_operation.wizard_description(), wizard_description)
+    
+    def test_operate(self):
+        self.data_operation.operate()
+        self.assertEqual(self.data_operation.get_percent_completion(), 1.0)
+        self.assertEqual(self.data_operation.get_status(), 'COMPLETED')
+        self.assertEqual(self.data_operation.get_output(), {'output_files': []})
+    
+    def test_generate_cache_key(self):
+        cache_key = self.data_operation.generate_cache_key()
+        self.assertIsInstance(cache_key, str)
+        self.assertGreater(len(cache_key), 0)
```

Comment:
It would be better to test the logic of creating the cache key, so in case we make a change we know that still creates the same keys we expect. So test the actual value of the key, not just that a key exists.